src/dataExplorer/components/DeleteScript.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {FC, useContext} from 'react'
+import {Button, ComponentColor, IconFont, Overlay} from '@influxdata/clockface'
+import './SaveAsScript.scss'
+import {CLOUD} from 'src/shared/constants'
+import {PersistanceContext} from 'src/dataExplorer/context/persistance'
+import {useHistory} from 'react-router-dom'
+import {useSelector, useDispatch} from 'react-redux'
+import {getOrg} from 'src/organizations/selectors'
+import {notify} from 'src/shared/actions/notifications'
+import {deleteScriptFail} from 'src/shared/copy/notifications/categories/scripts'
+
+let deleteScript
+
+if (CLOUD) {
+  deleteScript = require('src/client/scriptsRoutes').deleteScript
+}
+
+interface Props {
+  onBack: () => void
+  onClose: () => void
+}
+
+const OpenScript: FC<Props> = ({onBack, onClose}) => {
+  const {resource} = useContext(PersistanceContext)
+  const history = useHistory()
+  const dispatch = useDispatch()
+  const org = useSelector(getOrg)
+
+  const handleDeleteScript = async () => {
+    if (resource.data?.id) {
+      try {
+        const resp = await deleteScript({scriptID: resource.data.id})
+
+        if (resp.status !== 204) {
+          throw new Error(resp.data.message)
+        }
+
+        history.replace(`/orgs/${org.id}/data-explorer/from/script/`)
```

Comment:
Just curious, what's the purpose of this line? When I test on my local, I saw the URL changed very quickly from `.../data-explorer/` to `.../data-explorer/from/script/` then back to `.../data-explorer/` after a script is deleted

Reply:
Is this loading another script resource? And that trickles into any of the state objects (session store, contexts, etc)?

Reply:
Actually, we need to do more clearing of state. See the example `clear()` method in SaveAsScript component.

Otherwise, things stick around after delete. Such as the results:

https://user-images.githubusercontent.com/10232835/196870011-999135e1-f35a-4d61-bf81-0d1ca0bedf20.mov

Reply:
Hey! Sorry for not seeing this earlier. I would like it implemented how I designed it because I think the filled-in button over-emphasizes it in the modal. Not a blocker, but if we could update it eventually to match, that'd be great! Thanks for catching that @wiedld. I can talk to the design team to get this incorporated into our design system so it's not just a one-off.